compiler/src/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp

```diff
@@ -143,25 +143,66 @@ static LogicalResult getPaddingDims(func::FuncOp funcOp,
   return success();
 }
 
-/// Default method to initialize the tiling options for fusion in IREE. These
-/// could be ovveridden by the command line options if specified.
-static FailureOr<linalg::LinalgTilingAndFusionOptions>
-getTileAndFuseOptionsFromConfig(func::FuncOp funcOp, int64_t tilingLevel) {
+/// Default method to get tile sizes for tile-and-fuse in IREE. These could be
+/// ovveridden by the command line options if specified.
+static LogicalResult getTileAndFuseOptionsFromConfig(
+    func::FuncOp funcOp, int64_t tilingLevel,
+    SmallVector<int64_t> &tileAndFuseSizes, SmallVector<int64_t> &tileOnlySizes,
+    SmallVector<int64_t> &tileInterchange) {
   if (tilingLevel == -1) {
-    return linalg::LinalgTilingAndFusionOptions();
+    return success();
   }
 
   FailureOr<Operation *> rootOp = getRootOp(funcOp);
   if (failed(rootOp) || !rootOp.value()) return failure();
+  auto tilingOp = cast<TilingInterface>(rootOp.value());
 
   iree_compiler::IREE::Codegen::LoweringConfigAttr loweringConfig =
-      iree_compiler::getLoweringConfig(rootOp.value());
+      iree_compiler::getLoweringConfig(tilingOp);
+  SmallVector<int64_t> tileSizes = loweringConfig.getTileSizeVals(tilingLevel);
+
+  auto iteratorTypes = tilingOp.getLoopIteratorTypes();
+  tileAndFuseSizes = SmallVector<int64_t>(iteratorTypes.size(), /*default=*/0);
+  tileOnlySizes = SmallVector<int64_t>(iteratorTypes.size(), /*default=*/0);
+  // Splits the tileSizes into two sizes vectors. We want to tile-and-fuse on
+  // parallel dims and tile-only on reduction dims.
+  for (size_t i = 0; i < tileSizes.size() && i < iteratorTypes.size(); ++i) {
+    if (iteratorTypes[i] == utils::IteratorType::parallel) {
+      tileAndFuseSizes[i] = tileSizes[i];
+    } else {
+      tileOnlySizes[i] = tileSizes[i];
+    }
+  }
 
-  linalg::LinalgTilingAndFusionOptions options;
-  options.tileSizes.assign(loweringConfig.getTileSizeVals(tilingLevel));
-  options.tileInterchange.assign(
-      loweringConfig.getTileInterchangeVals(tilingLevel));
-  return options;
+  tileInterchange = loweringConfig.getTileInterchangeVals(tilingLevel);
+
+  return success();
+}
+
+static void populateTileAndFuseOptions(
```

Comment:
I think i'd rather just inline the function at the call site. That would avoid return by reference, and checking for tile size computation function being a `nullptr`. You can just use `anyNonZeros` to decide if tile + fuse or tiling needs to be run.

Reply:
Done.